jenkins-cucumber-test-collector/src/main/java/com/capitalone/dashboard/collector/DefaultJenkinsClient.java

```diff
@@ -179,10 +306,43 @@ private JSONArray getJsonArray(JSONObject json, String key) {
 
     private String getJson(String baseUrl, String endpoint) {
         String url = StringUtils.removeEnd(baseUrl, "/") + endpoint;
-        return rest.getForObject(URI.create(url), String.class);
+        ResponseEntity<String> result = null;
+        try {
+            result = makeRestCall(url);
+            return result.getBody();
+        } catch (MalformedURLException mfe) {
+            LOG.error("malformed url" + mfe.getStackTrace());
+        }
+        return "";
     }
 
-    private String getCucumberJson(String buildUrl, String artifactRelativePath) {
-        return getJson(buildUrl, "/artifact/" + artifactRelativePath);
+    protected ResponseEntity<String> makeRestCall(String sUrl) throws MalformedURLException {
+        URI thisuri = URI.create(sUrl);
+        String userInfo = thisuri.getUserInfo();
+
+        //get userinfo from URI or settings (in spring properties)
+        if (StringUtils.isEmpty(userInfo) && (this.settings.getUsername() != null) && (this.settings.getApiKey() != null)) {
+            userInfo = this.settings.getUsername() + ":" + this.settings.getApiKey();
+        }
+        // Basic Auth only.
+        if (StringUtils.isNotEmpty(userInfo)) {
+            return rest.exchange(thisuri, HttpMethod.GET,
+                    new HttpEntity<>(createHeaders(userInfo)),
+                    String.class);
+        } else {
```

Comment:
```if (StringUtils.isNotEmpty(userInfo)) {
            return rest.exchange(thisuri, HttpMethod.GET,
                    new HttpEntity<>(createHeaders(userInfo)),
                    String.class);
}
return rest.exchange(thisuri, HttpMethod.GET, null, String.class);
```
the else is not necessary, it will either return from inside the if or it will return..